// @tatumio/tatum/dist/src/util/network.utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkUtils = void 0;
const dto_1 = require("../dto");
const process_1 = __importDefault(require("process"));
exports.NetworkUtils = {
    getNetworkMetadata: (network) => {
        return dto_1.NETWORK_METADATA[network];
    },
    isTestnet(network) {
        return exports.NetworkUtils.getNetworkMetadata(network).testnet;
    },
    getChainId: (network) => {
        const chainId = dto_1.NETWORK_METADATA[network]?.chainId;
        if (!chainId) {
            throw new Error(`Tatum Network to ChainId for network ${network} does not exist`);
        }
        return chainId;
    },
    isAlternateTestnet: (network) => {
        const metadata = exports.NetworkUtils.getNetworkMetadata(network);
        return metadata.testnet && !metadata.defaultTestnet;
    },
    getV4ApiKeyForNetwork: (network) => exports.NetworkUtils.isTestnet(network) ? process_1.default.env.V4_API_KEY_TESTNET : process_1.default.env.V4_API_KEY_MAINNET
};
