// @tatumio/tatum/dist/src/e2e/rpc/evm/evm.rpc.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const dto_1 = require("../../../dto");
const evm_e2e_utils_1 = require("./evm.e2e.utils");
const testNetworks = [
    { network: dto_1.Network.CELO },
    { network: dto_1.Network.CELO_ALFAJORES },
    { network: dto_1.Network.ARBITRUM_ONE },
    { network: dto_1.Network.ARBITRUM_NOVA, apiKey: process.env.V3_API_KEY_MAINNET },
    {
        network: dto_1.Network.ARBITRUM_NOVA_TESTNET,
        apiKey: process.env.V3_API_KEY_TESTNET,
    },
    { network: dto_1.Network.HORIZEN_EON },
    // { network: Network.HORIZEN_EON_GOBI },
    { network: dto_1.Network.CHILIZ },
    { network: dto_1.Network.BINANCE_SMART_CHAIN },
    {
        network: dto_1.Network.BINANCE_SMART_CHAIN_TESTNET,
        apiKey: process.env.V3_API_KEY_TESTNET,
    },
    { network: dto_1.Network.FLARE },
    { network: dto_1.Network.FLARE_SONGBIRD },
    { network: dto_1.Network.FLARE_COSTON },
    { network: dto_1.Network.FLARE_COSTON_2 },
    { network: dto_1.Network.ETHEREUM },
    { network: dto_1.Network.ETHEREUM_SEPOLIA },
    { network: dto_1.Network.ETHEREUM_HOLESKY },
    { network: dto_1.Network.FANTOM },
    { network: dto_1.Network.FANTOM_TESTNET, apiKey: process.env.V3_API_KEY_TESTNET },
    { network: dto_1.Network.ETHEREUM_CLASSIC },
    { network: dto_1.Network.POLYGON },
    { network: dto_1.Network.POLYGON_MUMBAI },
    // { network: Network.OPTIMISM },
    { network: dto_1.Network.HAQQ },
    { network: dto_1.Network.HAQQ_TESTNET },
    {
        network: dto_1.Network.TRON,
        data: {
            estimateGas: {
                from: '0x41F0CC5A2A84CD0F68ED1667070934542D673ACBD8',
                to: '0x4170082243784DCDF3042034E7B044D6D342A91360',
                gas: '0x01',
                gasPrice: '0x8c',
                value: '0x01',
                data: '0x70a08231000000000000000000000041f0cc5a2a84cd0f68ed1667070934542d673acbd8',
            },
        },
    },
    {
        network: dto_1.Network.TRON_SHASTA,
        skipEstimateGas: true,
        apiKey: process.env.V3_API_KEY_TESTNET,
    },
    {
        network: dto_1.Network.KLAYTN,
    },
    {
        network: dto_1.Network.KLAYTN_BAOBAB,
        apiKey: process.env.V4_API_KEY_TESTNET,
    },
    {
        network: dto_1.Network.AVALANCHE_C,
    },
    {
        network: dto_1.Network.AVALANCHE_C_TESTNET,
        apiKey: process.env.V3_API_KEY_TESTNET,
    },
    {
        network: dto_1.Network.XINFIN,
    },
    // { network: Network.CRONOS },
    { network: dto_1.Network.CRONOS_TESTNET, apiKey: process.env.V3_API_KEY_TESTNET },
    { network: dto_1.Network.BASE },
];
describe.each(testNetworks)('RPC EVM', (testNetwork) => {
    const { network } = testNetwork;
    describe(network, () => {
        evm_e2e_utils_1.EvmE2eUtils.e2e(testNetwork);
    });
});
